src/App.js

```diff
@@ -1,26 +1,126 @@
 import React from 'react';
-import logo from './logo.svg';
+import FoodBox from './components/FoodBox';
+import AddFood from './components/AddFood';
+import Search from './components/Search';
+
+import foods from './foods.json';
+
+import 'bulma/css/bulma.css';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    cloneFoods: foods,
+    addForm: false,
+    todaysFoods: [],
+    totalCal: 0,
+  };
+
+  // Form handler //
+  handleAdd = (e) => {
+    e.preventDefault();
+    let name = e.target.addFoodName.value;
+    let image = e.target.addFoodImage.value;
+    let calories = e.target.addFoodCalories.value;
+
+    let newFood = {
+      name: name,
+      image: image,
+      calories: calories,
+      quantity: 0,
+    };
+    this.setState({
+      cloneFoods: [...foods, newFood],
+      addForm: false,
+    });
+  };
+
+  handleForm = () => {
+    this.setState({
+      addForm: true,
+    });
+  };
+
+  // Search handler //
+  handleSearch = (e) => {
+    e.preventDefault();
+    const { cloneFoods } = this.state;
+    let searchInput = e.target.value.toLowerCase();
+    let results = cloneFoods.filter((food) => {
+      return food.name.toLowerCase().includes(searchInput);
+    });
+    this.setState({
+      cloneFoods: results,
+    });
+  };
+
+  // Today's food handler //
+  handleTodaysFood = (item, qty) => {
+    let duplicate = false;
+    const { todaysFoods, totalCal } = this.state;
+    if (qty !== 0) {
+      let newFood = {
+        name: item.name,
+        calories: item.calories,
+        quantity: qty,
+      };
+
+      let newFoods = todaysFoods.map((food) => {
+        if (food.name == item.name) {
+          food = {
+            name: food.name,
+            calories:
+              newFood.calories * (parseInt(food.quantity) + parseInt(qty)),
+            quantity: parseInt(food.quantity) + parseInt(qty),
+          };
+          duplicate = true;
+          return food;
+        }
+        return food;
+      });
+
+      this.setState({
+        todaysFoods: duplicate ? [...newFoods] : [...newFoods, newFood],
+        totalCal: totalCal + newFood.quantity * newFood.calories,
+      });
+      duplicate = false;
+    }
+  };
+
+  handleDelete = (itemName) => {
+    const { todaysFoods } = this.state;
+    let filteredFoods = todaysFoods.filter((food) => {
+      return itemName !== food.name;
+    });
```

Comment:
to update the calories amount after you delete an element you can run a new .reduce on all the elements from todaysFood and then update the totalCal state.